### PR TITLE
Apply clippy suggestions from Rust 1.67

### PR DESCRIPTION
**Pull request by dzvon**

### CODE REVIEW

File: perf/src/bin/perf_server.rs [hunk 1]
```diff
@@ -50,7 +50,7 @@ async fn main() {
 
 async fn run(opt: Opt) -> Result<()> {
     let (key, cert) = match (&opt.key, &opt.cert) {
-        (&Some(ref key), &Some(ref cert)) => {
+        (Some(key), Some(cert)) => {
```
Comment (Ralith) [inline perf/src/bin/perf_server.rs:53]: This doesn't seem to be fixing a lint. If it was, I would report a bug, as the code was intended as written. Match "ergonomics" impair readability.

File: quinn-proto/src/connection/mod.rs [hunk 2]
```diff
@@ -1338,7 +1337,9 @@ impl Connection {
         let loss_delay = cmp::max(rtt.mul_f32(self.config.time_threshold), TIMER_GRANULARITY);
 
         // Packets sent before this time are deemed lost.
-        let lost_send_time = now - loss_delay;
+        let lost_send_time = now
+            .checked_sub(loss_delay)
+            .expect("overflow when subtracting duration from instant");
```
Comment (Ralith) [inline quinn-proto/src/connection/mod.rs:1342]: Technically this is an underflow, and in any case it's unlikely enough that an `unwrap()` is fine.

**Review by Ralith — CHANGES_REQUESTED** [review]

File: perf/src/bin/perf_server.rs (hunk 1 above)
Comment (dzvon) [reply perf/src/bin/perf_server.rs:53]:
```
error: dereferencing a tuple pattern where every element takes a reference
  --> perf/src/bin/perf_server.rs:53:10
   |
53 |         (&Some(ref key), &Some(ref cert)) => {
   |          ^^^^^^^^^^^^^^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_borrowed_reference
   = note: `-D clippy::needless-borrowed-reference` implied by `-D warnings`
help: try removing the `&` and `ref` parts
   |
53 -         (&Some(ref key), &Some(ref cert)) => {
53 +         (Some(key), &Some(ref cert)) => {
   |

error: dereferencing a tuple pattern where every element takes a reference
  --> perf/src/bin/perf_server.rs:53:26
   |
53 |         (&Some(ref key), &Some(ref cert)) => {
   |                          ^^^^^^^^^^^^^^^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_borrowed_reference
help: try removing the `&` and `ref` parts
   |
53 -         (&Some(ref key), &Some(ref cert)) => {
53 +         (&Some(ref key), Some(cert)) => {
   |

error: could not compile `perf` due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
error: could not compile `perf` due to 2 previous errors
```

This is the error output of `cargo clippy --all-targets -- -D warnings`.

Comment (Ralith) [reply perf/src/bin/perf_server.rs:53]: Thanks, opened https://github.com/rust-lang/rust-clippy/issues/10245. For now please allowlist the lint with a reference to that issue.

**Comment by djc** [comment]
I forgot that this existed and accidentally reimplemented it in #1499. Thanks for your work!

**Pull request closed without merging**